course/quickdrill.php

```diff
@@ -546,10 +549,13 @@ function makelink() {
 	 var val = document.getElementById("val").value;
 	 if (mode!='none' && val=='') { alert("need to specify N"); return false;}
 	 var url = baseaddr + '?id=' + id + '&sa='+sa;
+	 if (!isNumeric(id) || (!isNumeric(cid) && cid != 'admin') || !isNumeric(val)) {

```

Comment:
probably excessive - maybe use encodeURIComponent on each of the params instead?